analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -295,9 +297,15 @@ private static ProgramState InitLocals(ControlFlowBranch branch, ProgramState st
         branch.IsConditionalSuccessor ? condition == constraint : condition != constraint;
 
     private static bool IsReachable(ControlFlowRegion region, ExceptionState thrown) =>
-        region.Kind == ControlFlowRegionKind.Finally
+        CatchesAll(region)
         || thrown == ExceptionState.UnknownException
-        || region.ExceptionType.Is(KnownType.System_Object)       // catch when (condition)
-        || region.ExceptionType.Is(KnownType.System_Exception)
         || thrown.Type.DerivesFrom(region.ExceptionType);
+
+    private static bool CatchesAll(ControlFlowRegion region) =>
+        region.Kind switch
+        {
+            ControlFlowRegionKind.FilterAndHandler => false,
```

Comment:
And what if we know that the filter is `true`? Is it too early to evaluate that by now?

Reply:
I have not looked into it. The precision could be improved in a different PR.